### PR TITLE
Add Checkin API and Badge Preview and Download Feature

### PR DESCRIPTION
**Pull request by Sak1012**
This PR let's organisers perform Checkin from Open-event-checkin with the new CheckIn API and also facilitates badge preview, printing and downloading functionalities at checkin.

## Summary by Sourcery

Add a new Checkin API and badge preview, printing, and downloading features to enhance the check-in process. Implement PDF rendering support and configure CORS headers for cross-origin requests.

New Features:
- Introduce a new Checkin API that allows organizers to perform check-ins from Open-event-checkin.
- Add badge preview, printing, and downloading functionalities at check-in.

Enhancements:
- Implement a PDFRenderer class to support PDF rendering in the badges plugin.
- Add CORS headers to the Nginx configuration to allow cross-origin requests from the check-in application.

Deployment:
- Update Nginx configuration to include CORS headers for the check-in application.

### CODE REVIEW

File: src/pretix/control/templates/pretixcontrol/pdf/index.html [hunk 1]
```diff
@@ -391,24 +391,24 @@ <h2>{% trans "Welcome to the PDF ticket editor!" %}</h2>
...
                         <span class="fa fa-qrcode"></span>
                         {% trans "QR code for Lead Scanning" %}
-                    </button>
+                    </button>-->
                     <button class="btn btn-default btn-block" id="editor-add-poweredby"
                             data-content="dark"
                             disabled>
                         <span class="fa fa-image"></span>
-                        {% trans "pretix Logo" %}
+                        {% trans "Eventyay Logo" %}
```
Comment (mariobehling) [inline src/pretix/control/templates/pretixcontrol/pdf/index.html]: eventyay Logo

**Review by mariobehling — CHANGES_REQUESTED** [review]
Please ensure tests are passing.

File: src/pretix/control/templates/pretixcontrol/pdf/index.html (hunk 1 above)
Comment (Sak1012) [reply src/pretix/control/templates/pretixcontrol/pdf/index.html]: Done 👍

**Comment by Sak1012** [comment]
> Please ensure tests are passing.

I have made the necessary modifications 👍

File: deployment/docker/nginx.conf [hunk 2]
```diff
@@ -47,6 +47,22 @@ http {
         server_name _;
         index index.php index.html;
         root /var/www;
+        
+        add_header 'Access-Control-Allow-Origin' 'https://checkin.eventyay.com' always;
```
Comment (mariobehling) [inline deployment/docker/nginx.conf]: Please change this into a variable. So we can use different URLs, e.g. for the development branch.

**Review by mariobehling — CHANGES_REQUESTED** [review]

File: src/pretix/settings.py [hunk 3]
```diff
@@ -373,6 +373,8 @@
 }
 
 MIDDLEWARE = [
+    'corsheaders.middleware.CorsMiddleware',
```
Comment (hongquan) [inline src/pretix/settings.py]: You should remove this middleware to test if the Nginx config works.

File: deployment/docker/nginx.conf (hunk 2 above)
Comment (hongquan) [reply deployment/docker/nginx.conf]: How about using wildcard `*` to allow all "origin"?

**Review by hongquan — CHANGES_REQUESTED** [review]

File: deployment/docker/nginx.conf (hunk 2 above)
Comment (mariobehling) [reply deployment/docker/nginx.conf]: Well, for partners such as Wikimedia we also want to deploy it to different domains.

File: src/pretix/settings.py (hunk 3 above)
Comment (Sak1012) [reply src/pretix/settings.py]: Done 👍

**Pull request merged**